common/signal_processing/src/butterworth.cpp

```diff
@@ -0,0 +1,307 @@
+    return;
+  }
+
+  cutoff_frequency_rad_sec = fc * 2.0 * M_PI;
+  sampling_frequency_hz = fs;
+}
+
+sOrderCutoff ButterworthFilter::getOrderCutOff() const
+{
+  return sOrderCutoff{order_, cutoff_frequency_rad_sec};
+}
+
+/**
+ * @brief Matlab equivalent : [b, a]  = butter(n, Wn, 's')
+ * */
+void ButterworthFilter::computeContinuousTimeTF(const bool & use_sampling_frequency)
+{
+  // First compute  the phase angles of the roots
+  computePhaseAngles();
+  computeContinuousTimeRoots(use_sampling_frequency);
+
+  continuous_time_denominator_ = poly(continuous_time_roots_);
+  continuous_time_numerator_ = std::pow(cutoff_frequency_rad_sec, order_);
+}
+
+void ButterworthFilter::computePhaseAngles()
+{
+  phase_angles_.resize(order_, 0.);
+
+  for (size_t i = 0; i < phase_angles_.size(); ++i) {
+    auto & x = phase_angles_.at(i);
+    x = M_PI_2 + (M_PI * (2.0 * static_cast<double>((i + 1)) - 1.0) / (2.0 * order_));
+  }
+}
+
+void ButterworthFilter::computeContinuousTimeRoots(const bool & use_sampling_frequency)
+{
+  continuous_time_roots_.resize(order_, {0.0, 0.0});
+
+  if (use_sampling_frequency) {
+    double const & Fc = (sampling_frequency_hz / M_PI) *
+                        tan(cutoff_frequency_rad_sec / (sampling_frequency_hz * 2.0));
+
+    for (size_t i = 0; i < continuous_time_roots_.size(); ++i) {
+      auto & x = continuous_time_roots_[i];
+      x = {
+        std::cos(phase_angles_[i]) * Fc * 2.0 * M_PI, std::sin(phase_angles_[i]) * Fc * 2.0 * M_PI};
+    }
+    return;
+  }
+
+  for (size_t i = 0; i < continuous_time_roots_.size(); ++i) {
+    auto & x = continuous_time_roots_[i];
+    x = {
+      cutoff_frequency_rad_sec * cos(phase_angles_[i]),
+      cutoff_frequency_rad_sec * sin(phase_angles_[i])};
+  }
+}
+std::vector<std::complex<double>> ButterworthFilter::poly(
+  std::vector<std::complex<double>> const & roots)
+{
+  std::vector<std::complex<double>> coefficients(roots.size() + 1, {0, 0});
+
+  int const n{static_cast<int>(roots.size())};
+
+  coefficients[0] = {1.0, 0.0};
+
+  for (int i = 0; i < n; i++) {
+    for (int j = i; j != -1; j--) {
+      coefficients[j + 1] = coefficients[j + 1] - (roots[i] * coefficients[j]);
+    }
+  }
+
+  return coefficients;
+}
+
+/**
+ *  @brief Prints the order and cut-off angular frequency (rad/sec) of the filter
+ * */
+
+void ButterworthFilter::printFilterContinuousTimeRoots() const
+{
+  std::stringstream stream;
+  stream << "\n The roots of the continuous-time filter Transfer Function's Denominator are : \n";
+
+  for (auto const & x : continuous_time_roots_) {
+    stream << std::fixed << std::setprecision(2) << std::real(x) << " j";
+
+    auto txt = std::imag(x) < 0 ? " - j " : " + j ";
+    stream << std::fixed << std::setprecision(2) << txt << std::abs(std::imag(x)) << " \n";
+  }
+
+  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "%s", stream.str().c_str());
+}
+void ButterworthFilter::printContinuousTimeTF() const
+{
+  auto const & n = order_;
```

Comment:
`const auto`

There are same code like `int const`, `double const`. Could you fix all of them please.